### PR TITLE
implement teachers popularity

### PR DESCRIPTION
**Pull request by subhangi2731**
## Related Issue 

- implement teachers popularity among users

Closes: #181

### Describe the changes you've made

Made changes in usercontroller.js , usermodel.js and card.js

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement in the existing features
- [ ] Other

## Mention any unusual behaviour of your code (Write NA if not)
NA

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] I have verified/tested my code by running it locally.

### CODE REVIEW

**Comment by subhangi2731** [comment]
@Namanl2001 please merge my pr I have done this

**Comment by subhangi2731** [comment]
@Namanl2001 please check and let me know and merge this pr it took a lot to me to do evrything

**Comment by subhangi2731** [comment]
@Namanl2001 i request u please check and merge my pr

File: src/components/card.jsx [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+    axios
+      .patch(`/users/update/${this.props.data._id}`, {
+        popularity: this.state.popularity+1
+      })
+      .then(response => {
+        console.log(response)
+      })
+      .catch(function (error) {
+        console.log(error);
+      });
```
Comment (Namanl2001) [inline src/components/card.jsx:22]: please handle `res` and `err` properly. And remove the console logs

Phal @ [this](https://itnext.io/javascript-error-handling-from-express-js-to-react-810deb5e5e28)

File: src/components/card.jsx [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+      clicked:true
+    })
+  };
+
   render() {
     return (
       <div className='box-container'>
         <div className='box-item'>
-          <div className='flip-box'>
+          <div className='flip-box' onClick={this.state.clicked?'':this.handleClick}>
```
Comment (Namanl2001) [inline src/components/card.jsx:32]: There is a hover effect on cards so this onclick will not work practically.
I'll suggest you to move this `onClick` event listener to the `click here to read more` button on the cards back.

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Small nits. Thanks for your patience

**Comment by tend2infinity** [comment]
I am taking over this PR since @subhangi2731 isn't active

**Comment by subhangi2731** [comment]
@tend2infinity why dont u create ur own pr why are u behind my pr I had health issues thatwhy I couldnt reply that doesnt mean I am not active and then u take over my pr @Namanl2001 as this is my pr and I have worked on this can I make changes and can u merge because I had send the pr before may 31 so I should get scores for this

**Comment by Namanl2001** [comment]
Yes, @subhangi2731 you can still make the required changes and we can have this merge.

**Comment by Namanl2001** [comment]
@subhangi2731 FYI your changes after tonight (i.e. 4/6/21 23:59 P.M. IST) will not be accepted. Hurry up!
Thanks

**Pull request closed without merging**